src/main/java/com/google/cloud/tools/app/impl/config/DefaultVersionsSelectionConfiguration.java

```diff
@@ -26,10 +26,20 @@
  */
 public class DefaultVersionsSelectionConfiguration implements VersionsSelectionConfiguration {
 
+  private String project;
   private Collection<String> versions;
   private String service;
 
   @Override
+  public String getProject() {
```

Comment:
I think we should create an abstract class called DefaultConfiguration that will have the getProject/setProject, and then just extend it here. Otherwise we'll be duplicating these accessor methods a lot.